pubsub/rocketmq/metadata.go

```diff
@@ -14,63 +14,157 @@ limitations under the License.
+
+	// Batch consumption size
+	ConsumeMessageBatchMaxSize int `mapstructure:"consumeMessageBatchMaxSize"`
+
+	// Concurrently max span offset.it has no effect on sequential consumption
+	ConsumeConcurrentlyMaxSpan int `mapstructure:"consumeConcurrentlyMaxSpan"`
+
+	// Max re-consume times. -1 means 16 times.
+	//
+	// If messages are re-consumed more than {@link #maxReconsumeTimes} before Success, it's be directed to a deletion
+	// queue waiting.
+	MaxReconsumeTimes int32  `mapstructure:"maxReconsumeTimes"`
+	AutoCommit        string `mapstructure:"autoCommit"`
+
+	// Maximum amount of time a message may block the consuming thread.
+	//
+	// RocketMQ Go Client does not support configuration in github.com/apache/rocketmq-client-go/v2 v2.1.1-rc2
+	ConsumeTimeout int `mapstructure:"consumeTimeout"`
+
+	// The socket timeout in milliseconds
+	ConsumerPullTimeout int `mapstructure:"consumerPullTimeout"`
+
+	// Message pull Interval
+	PullInterval int `mapstructure:"pullInterval"`
+
+	// Batch pull size
+	PullBatchSize int32 `mapstructure:"pullBatchSize"`
+
+	// Flow control threshold on queue level, each message queue will cache at most 1000 messages by default,
+	// Consider the {PullBatchSize}, the instantaneous value may exceed the limit
+	//
+	// RocketMQ Go Client does not support configuration in github.com/apache/rocketmq-client-go/v2 v2.1.1-rc2
+	PullThresholdForQueue int64 `mapstructure:"pullThresholdForQueue"`
+
+	// Flow control threshold on topic level, default value is -1(Unlimited)
+	//
+	// The value of {@code pullThresholdForQueue} will be overwrote and calculated based on
+	// {@code pullThresholdForTopic} if it is't unlimited
+	//
+	// For example, if the value of pullThresholdForTopic is 1000 and 10 message queues are assigned to this consumer,
+	// then pullThresholdForQueue will be set to 100
+	//
+	// RocketMQ Go Client does not support configuration in github.com/apache/rocketmq-client-go/v2 v2.1.1-rc2
+	PullThresholdForTopic int64 `mapstructure:"pullThresholdForTopic"`
+
+	// RocketMQ Go Client does not support configuration in github.com/apache/rocketmq-client-go/v2 v2.1.1-rc2
+	PullThresholdSizeForQueue int `mapstructure:"pullThresholdSizeForQueue"`
+
+	// Limit the cached message size on topic level, default value is -1 MiB(Unlimited)
+	//
+	// The value of {@code pullThresholdSizeForQueue} will be overwrote and calculated based on
+	// {@code pullThresholdSizeForTopic} if it is't unlimited
+	//
+	// For example, if the value of pullThresholdSizeForTopic is 1000 MiB and 10 message queues are
+	// assigned to this consumer, then pullThresholdSizeForQueue will be set to 100 MiB
+	//
+	// RocketMQ Go Client does not support configuration in github.com/apache/rocketmq-client-go/v2 v2.1.1-rc2
+	PullThresholdSizeForTopic int    `mapstructure:"pullThresholdSizeForTopic"`
+	ContentType               string `mapstructure:"content-type"` // msg's content-type
+	SendMsgTimeout            int    `mapstructure:"sendMsgTimeout"`
```

Comment:
This one too seems to be `sendTimeOut` renamed, which is a breaking change

Reply:
ok,  I will use the following code to achieve compatibility
func parseRocketMQMetaData in metadata.go: 
if _, ok := metadata.Properties["sendMsgTimeout"]; !ok {
			if v, ok := metadata.Properties["sendTimeOut"]; ok {
				if sendMsgTimeout, e := strconv.Atoi(v); e == nil {
					rMetaData.SendMsgTimeout = sendMsgTimeout
				}
			}
		}